storage/fs.go

```diff
@@ -85,27 +86,36 @@ func newFileSystemManager(
 		opts:     opts,
 		jitter:   jitter,
 		status:   fileOpNotStarted,
+		enabled:  true,
 	}, nil
 }
 
-func (m *fileSystemManager) ShouldRun(t time.Time) bool {
-	// If we haven't bootstrapped yet, no actions necessary.
-	if !m.database.IsBootstrapped() {
-		return false
-	}
+func (m *fileSystemManager) Disable() bool {
+	m.Lock()
+	isRunning := m.status == fileOpInProgress
+	m.enabled = false
+	m.Unlock()
+	return isRunning
+}
 
-	m.RLock()
-	defer m.RUnlock()
+func (m *fileSystemManager) Enable() {
+	m.Lock()
+	m.enabled = true
+	m.Unlock()
+}
 
-	// If we are in the middle of performing file operations, bail early.
-	return m.status != fileOpInProgress
+func (m *fileSystemManager) IsRunning() bool {
+	m.RLock()
+	running := m.status == fileOpInProgress
+	m.RUnlock()
+	return running
 }
 
-func (m *fileSystemManager) Run(t time.Time, async bool) {
+func (m *fileSystemManager) Run(t time.Time, async bool, force bool) bool {
```

Comment:
Can we perhaps make these booleans enums so it's clear what the params are when we use them from calling code? i.e. something like:
```go
type runType int
const (
  syncRun runType = iota
  asyncRun
)
type forceType int
const (
  noForce forceType = iota
  force 
)

// calling
fsm.Run(t, syncRun, noForce)
```